### PR TITLE
Build results from 2020-12-27

### PR DESCRIPTION
**Pull request by BillyONeal**
Build results from 2020-12-27 https://dev.azure.com/vcpkg/public/_build/results?buildId=47034

PASSING, REMOVE FROM FAIL LIST: capnproto:x64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: capnproto:x64-windows-static (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: capnproto:x86-windows (.\scripts\ci.baseline.txt)

These were fixed by the update to VS2019 version 16.8. Other capnproto skips moved into "supports" because:
capnp.lib(serialize-packed.c++.obj) : error LNK2019: unresolved external symbol __popcnt referenced in function "int __cdecl kj::popCount(unsigned int)" (?popCount@kj@@YAHI@Z)
i.e. the port wants popcnt to exist and it doesn't, so this isn't an artifact of our CI.

PASSING, REMOVE FROM FAIL LIST: libepoxy:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: libsamplerate:arm-uwp (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: libsamplerate:arm64-windows (.\scripts\ci.baseline.txt)
PASSING, REMOVE FROM FAIL LIST: nanogui:arm64-windows (.\scripts\ci.baseline.txt)

Not sure why these got fixed.

REGRESSION: atk:x86-windows. If expected, add atk:x86-windows=fail to .\scripts\ci.baseline.txt
REGRESSION: dcmtk:x86-windows. If expected, add dcmtk:x86-windows=fail to .\scripts\ci.baseline.txt.
REGRESSION: libsigcpp:x86-windows. If expected, add libsigcpp:x86-windows=fail to .\scripts\ci.baseline.txt.

These all had various timeouts and TLS errors connecting ftp.gnome.org; I updated all ports targeting this server to use HTTPs.

REGRESSION: chartdir:x64-osx. If expected, add chartdir:x64-osx=fail to .\scripts\ci.baseline.txt.

Upstream sources broke:

```
-- Note: chartdir only supports dynamic library linkage. Building dynamic library.
-- Downloading https://www.advsofteng.net/chartdir_cpp_mac.tar.gz...
```

REGRESSION: leaf:arm-uwp. If expected, add leaf:arm-uwp=fail to .\scripts\ci.baseline.txt.
REGRESSION: leaf:arm64-windows. If expected, add leaf:arm64-windows=fail to .\scripts\ci.baseline.txt.
REGRESSION: leaf:x64-linux. If expected, add leaf:x64-linux=fail to .\scripts\ci.baseline.txt.
REGRESSION: leaf:x64-osx. If expected, add leaf:x64-osx=fail to .\scripts\ci.baseline.txt.
REGRESSION: leaf:x64-uwp. If expected, add leaf:x64-uwp=fail to .\scripts\ci.baseline.txt
REGRESSION: leaf:x64-windows-static. If expected, add leaf:x64-windows-static=fail to .\scripts\ci.baseline.txt.
REGRESSION: leaf:x64-windows. If expected, add leaf:x64-windows=fail to .\scripts\ci.baseline.txt
REGRESSION: leaf:x86-windows. If expected, add leaf:x86-windows=fail to .\scripts\ci.baseline.txt.

```
Starting package 759/1414: leaf:x86-windows
Building package leaf[core]:x86-windows...
-- Downloading https://github.com/zajo/leaf/archive/0.2.2.tar.gz...
-- Extracting source D:/downloads/zajo-leaf-0.2.2.tar.gz
-- Using source at D:/buildtrees/leaf/src/0.2.2-26417cde9d.clean
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: D:/packages/leaf_x86-windows/share/leaf/copyright
-- Performing post-build validation
-- Performing post-build validation done
Uploaded binaries to 1 HTTP remotes.
Building package leaf[core]:x86-windows... done
Installing package leaf[core]:x86-windows...
The following files are already installed in D:/installed/x86-windows and are in conflict with leaf:x86-windows

Installed by boost-leaf:x86-windows
    include/boost/leaf/capture.hpp
    include/boost/leaf/common.hpp
    include/boost/leaf/context.hpp
    include/boost/leaf/detail/all.hpp
    include/boost/leaf/detail/demangle.hpp
    include/boost/leaf/detail/function_traits.hpp
    include/boost/leaf/detail/mp11.hpp
    include/boost/leaf/detail/optional.hpp
    include/boost/leaf/detail/print.hpp
    include/boost/leaf/error.hpp
    include/boost/leaf/exception.hpp
    include/boost/leaf/result.hpp

Elapsed time for package leaf:x86-windows: 2.931 s
```

Caused by the update to Boost 1.75.0. Because this port is touching include/boost, I'm interpreting that as "the author intends users to use it as a Boost library", and have deleted this port in favor of the Boost version.

REGRESSION: magic-get:arm-uwp. If expected, add magic-get:arm-uwp=fail to .\scripts\ci.baseline.txt.
REGRESSION: magic-get:arm64-windows. If expected, add magic-get:arm64-windows=fail to .\scripts\ci.baseline.txt.
REGRESSION: magic-get:x64-linux. If expected, add magic-get:x64-linux=fail to .\scripts\ci.baseline.txt.
REGRESSION: magic-get:x64-osx. If expected, add magic-get:x64-osx=fail to .\scripts\ci.baseline.txt.
REGRESSION: magic-get:x64-uwp. If expected, add magic-get:x64-uwp=fail to .\scripts\ci.baseline.txt.
REGRESSION: magic-get:x64-windows-static. If expected, add magic-get:x64-windows-static=fail to .\scripts\ci.baseline.txt
REGRESSION: magic-get:x64-windows. If expected, add magic-get:x64-windows=fail to .\scripts\ci.baseline.txt.
REGRESSION: magic-get:x86-windows. If expected, add magic-get:x86-windows=fail to .\scripts\ci.baseline.txt.

This library got accepted as boost-pfr. Changing to use the boost port under the same rationale as leaf.

```
Building package magic-get[core]:x86-windows...
-- Downloading https://github.com/apolukhin/magic_get/archive/abb467c0e22a83bf75a46a9e6610370fabfc39af.tar.gz...
-- Extracting source D:/downloads/apolukhin-magic_get-abb467c0e22a83bf75a46a9e6610370fabfc39af.tar.gz
-- Using source at D:/buildtrees/magic-get/src/0fabfc39af-4ec29c417c.clean
-- Using cached D:/downloads/boost_license_1_0.txt
-- Installing: D:/packages/magic-get_x86-windows/share/magic-get/copyright
-- Performing post-build validation
-- Performing post-build validation done
Uploaded binaries to 1 HTTP remotes.
Building package magic-get[core]:x86-windows... done
Installing package magic-get[core]:x86-windows...
The following files are already installed in D:/installed/x86-windows and are in conflict with magic-get:x86-windows

Installed by boost-pfr:x86-windows
    include/boost/pfr.hpp
    include/boost/pfr/detail/cast_to_layout_compatible.hpp
    include/boost/pfr/detail/config.hpp
    include/boost/pfr/detail/core14_classic.hpp
    include/boost/pfr/detail/core14_loophole.hpp
    include/boost/pfr/detail/core17.hpp
    include/boost/pfr/detail/core17_generated.hpp
    include/boost/pfr/detail/detectors.hpp
    include/boost/pfr/detail/fields_count.hpp
    include/boost/pfr/detail/for_each_field_impl.hpp
    include/boost/pfr/detail/functional.hpp
    include/boost/pfr/detail/io.hpp
    include/boost/pfr/detail/make_flat_tuple_of_references.hpp
    include/boost/pfr/detail/make_integer_sequence.hpp
    include/boost/pfr/detail/offset_based_getter.hpp
    include/boost/pfr/detail/rvalue_t.hpp
    include/boost/pfr/detail/sequence_tuple.hpp
    include/boost/pfr/detail/size_array.hpp
    include/boost/pfr/detail/stdtuple.hpp
    include/boost/pfr/detail/tie_from_structure_tuple.hpp

Elapsed time for package magic-get:x86-windows: 1.472 s
```

REGRESSION: python3:x64-windows-static. If expected, add python3:x64-windows-static=fail to .\scripts\ci.baseline.txt.
REGRESSION: python3:x64-windows. If expected, add python3:x64-windows=fail to .\scripts\ci.baseline.txt.
REGRESSION: python3:x86-windows. If expected, add python3:x86-windows=fail to .\scripts\ci.baseline.txt.

Probably caused by #14891

```
     3>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\include\cstdint(21,1): error C2061: syntax error: identifier 'std' [D:\buildtrees\python3\x86-windows-rel\v3.9.0-664bf2a259.clean\PCbuild\pythoncore.vcxproj]
     3>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\include\cstdint(21,1): error C2059: syntax error: ';' [D:\buildtrees\python3\x86-windows-rel\v3.9.0-664bf2a259.clean\PCbuild\pythoncore.vcxproj]
```

These files are C++ files getting built as C. mpdecimal.h seems to be the root of it but it has:

```
#ifdef __cplusplus
  #include <cinttypes>
  #include <climits>
  #include <cstdint>
  #include <cstdio>
  #include <cstdlib>
extern "C" {
#else
  #include <inttypes.h>
  #include <limits.h>
  #include <stdint.h>
  #include <stdio.h>
  #include <stdlib.h>
#endif
```

so I'm not sure exactly what is up right now....

@Hoikas any ideas?

### CODE REVIEW

**Comment by Hoikas** [comment]
@BillyONeal That's strange. Python 3.9 should build correctly with VS 2015 or higher. I'll fire up a VS 2017 virtual machine and look soon.

**Comment by BillyONeal** [comment]
@Hoikas to clarify, when I try to build this locally it works fine, it's only broken in CI; I suspect it's some hidden conflict with another port. But I didn't see where `/TC` is coming from in any of Python's build scripts....

File: ports/chartdir/portfile.cmake [hunk 1]
```diff
@@ -28,15 +28,15 @@ elseif(VCPKG_TARGET_IS_WINDOWS)
...
+
     set(CHARTDIR_LIB "chartdir60.lib")
 
 elseif(VCPKG_TARGET_IS_OSX)
 
     vcpkg_download_distfile(ARCHIVE_FILE
         URLS "https://www.advsofteng.net/chartdir_cpp_mac.tar.gz"
-        FILENAME "chartdir_cpp_mac-6.3.1.tar.gz"
-        SHA512 d7364b42ed7c5dd2f89ed8692ed0df1ec952216ba24f8afa8e039641f3296f23c9ff6391e3e0dcd3bfdb9e6898c21a8281aa2803fbee33c7197c1df271a0edf7
+        FILENAME "chartdir_cpp_mac-6.3.1.2020-12-28.tar.gz"
```
Comment (ras0219) [inline ports/chartdir/portfile.cmake:38]:
```suggestion
        FILENAME "chartdir_cpp_mac-6.3.1.51483c1975.tar.gz"
```

File: ports/leaf/CONTROL [hunk 2]
```diff
@@ -1,3 +1,4 @@
 Source: leaf
-Version: 0.2.2
-Description: Lightweight error augmentation framework 
\ No newline at end of file
+Version: removed
```
Comment (ras0219) [inline ports/leaf/CONTROL:2]: With versioning, this would need to be `0.2.2#1` to enable graceful constraint transition.

**Review by ras0219 — APPROVED** [review]

File: ports/leaf/CONTROL (hunk 2 above)
Comment (BillyONeal) [reply ports/leaf/CONTROL:2]: Hmmm... doesn't that make it a lie though?

Comment (BillyONeal) [reply ports/leaf/CONTROL:2]: @vicroms Maybe we want some kind of way to say "this one is replaced with that one" in version-space

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM.

**Comment by BillyONeal** [comment]
@Hoikas OK I got the python3 regression to repro consistently locally by running CI on my local box overrnight; will keep you posted

**Comment by BillyONeal** [comment]
@Hoikas Filed: https://github.com/microsoft/vcpkg/issues/15377

**Comment by Hoikas** [comment]
Great work! I was at a loss on this one.

**Pull request merged**